src/openforms/registrations/contrib/objects_api/tests/test_backend.py

```diff
@@ -1253,3 +1257,127 @@ def test_submission_with_payment(self, m):
                 "public_order_ids": [],
             },
         )
+
+
+@requests_mock.Mocker()
+@freeze_time()
+class ObjectsAPIBackendV2Tests(TestCase):
+
+    def setUp(self):
+        super().setUp()
+
+        config = ObjectsAPIConfig(
+            objects_service=ServiceFactory.build(
+                api_root="https://objecten.nl/api/v1/",
+                api_type=APITypes.orc,
+            ),
+            drc_service=ServiceFactory.build(
+                api_root="https://documenten.nl/api/v1/",
+                api_type=APITypes.drc,
+            ),
+        )
+
+        config_patcher = patch(
+            "openforms.registrations.contrib.objects_api.models.ObjectsAPIConfig.get_solo",
+            return_value=config,
+        )
+        self.mock_get_config = config_patcher.start()
+        self.addCleanup(config_patcher.stop)
+
+    def test_submission_with_objects_api_v2(self, m: requests_mock.Mocker):
+        submission = SubmissionFactory.from_components(
+            [
+                {
+                    "key": "myTextField",
+                    "type": "textfield",
+                },
+                {"key": "myNumberField", "type": "number"},
+                {
+                    "key": "myMapField",
+                    "type": "map",
+                },
+            ],
+            completed=True,
+            submitted_data={
+                "myTextField": "textfield value",
+                "myNumberField": 12.5,
+                "myMapField": [52.36673378967122, 4.893164274470299],
+            },
+        )
+
+        v2_options: RegistrationOptionsV2 = {
+            "version": 2,
+            "objecttype": "https://objecttypen.nl/api/v1/objecttypes/1",
+            "objecttype_version": 1,
+            "variables_mapping": [
+                {
+                    "variable_key": "myTextField",
+                    "target_path": ["path.to", "target"],
+                },
+                {"variable_key": "myNumberField", "target_path": ["other", "path"]},
+                {
+                    "variable_key": "now",
+                    "target_path": ["now_path"],
+                },
+            ],
+            "geometry_variable_key": "myMapField",
+        }
+
+        m.post(
+            "https://objecten.nl/api/v1/objects",
+            status_code=201,
+            json=get_create_json,
+        )
```

Comment:
can we please use docker-compose + VCR instead of these manual mocks?

I'd also like to see the tests for v1 and v2 in separate modules, there's nearly 1400 lines in this file and I can't put them side by side to compare if the same scope of functionality is covered